components/ToDoList/index.js

```diff
@@ -1,20 +1,79 @@
-import { TodoItem, TodoList, Checkbox } from "./styles";
+import React, { useState } from "react";
+import styled from "styled-components";
+import { useRouter } from "next/router";
+import { uid } from "uid";
 
-export default function ToDoList() {
-  const todoListData = [
-    { id: 1, text: "Water the plants" },
-    { id: 2, text: "Clean the houses" },
-    { id: 3, text: "collect the eggs" },
-  ];
+const TodoItemContainer = styled.li`
+  margin: 5px;
+  display: flex;
+  align-items: center;
+`;
+
+const TodoText = styled.span`
+  flex: 1;
+`;
+
+const DeleteButton = styled.button`
+  background-color: #ff0000;
+  color: #ffffff;
+  border: none;
+  cursor: pointer;
+  padding: 5px 10px;
+  border-radius: 5px;
+`;
+
+function TodoItem({ todo, onDelete }) {
+  return (
+    <TodoItemContainer>
+      <TodoText>{todo}</TodoText>
+      <DeleteButton onClick={onDelete}>Delete</DeleteButton>
+    </TodoItemContainer>
+  );
+}
+
+export default function TodoList({ listItems, setListItems }) {
+  const router = useRouter();
+  const [todos, setTodos] = useState([]);
+  const [newTodo, setNewTodo] = useState("");
+
+  const handleInputChange = (event) => {
+    setNewTodo(event.target.value);
+  };
+
+  function handleAdd(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    console.log(data);
+    setListItems([...listItems, { id: uid(), ...data }]);
+    console.log("listItems:", listItems);
+    router.push("/");
+  }
+
+  const handleDeleteTodo = (index) => {
+    const newTodos = todos.filter((_, i) => i !== index);
+    setTodos(newTodos);
+  };
 
   return (
-    <TodoList>
-      {todoListData.map((todo) => (
-        <TodoItem key={todo.id}>
-          <Checkbox />
-          {todo.text}
-        </TodoItem>
-      ))}
-    </TodoList>
+    <div>
+      <h1>Todo List</h1>
+      <form onSubmit={handleAdd}>
+        <input type="text" name="text" id="text" />
+        <label htmlFor="text"></label>
+        <button type="submit">Add</button>
+      </form>
+      <ul>
+        {todos.map((todo, index) => (
+          <TodoItem
+            key={index}
```

Comment:
Passing the `index` as the `key` is not recommended since it's **essentially the same as not providing a key at all**. Make use of the `id` in your `todo` parameter and pass it to the `key` here.